crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1450,14 +1459,16 @@ impl<RpcMiddleware> RpcServerConfig<RpcMiddleware> {
     /// Returns the [`RpcServerHandle`] with the handle to the started servers.
     pub async fn start(self, modules: &TransportRpcModules) -> Result<RpcServerHandle, RpcError>
     where
-        RpcMiddleware: Layer<RpcRequestMetricsService<RpcService>> + Clone + Send + 'static,
-        for<'a> <RpcMiddleware as Layer<RpcRequestMetricsService<RpcService>>>::Service:
+        RpcMiddleware: Layer<RpcRequestMetricsService<RpcRequestRateLimitingService<RpcService>>> + Clone + Send + 'static,
```

Comment:
I'd like to flip the order here because otherwise the call metrics would also include the time we wait for the rate limit